ui-modules/blueprint-composer/app/components/quick-fix/quick-fix.js

```diff
@@ -55,6 +55,31 @@ export function computeQuickFixes(blueprintService, allIssues) {
             computeQuickFixesForIssue(issue, issue.entity, blueprintService, v.quickFixes)
         });
     });
+
+    allIssues.warnings.byMessage = {};
+    Object.values(allIssues.warnings.byEntity).forEach(list => {
+        list.forEach(issue => {
+            let key = issue.group+":"+issue.ref+":"+issue.message;
+            let v = allIssues.warnings.byMessage[key];
+            if (!v) {
+                v = allIssues.warnings.byMessage[key] = {
+                    group: issue.group,
+                    ref: issue.ref,
+                    message: issue.message,
+                    issues: [],
+                    quickFixes: {},
+                };
+            }
+
+            let issueO = {
+                issue,
+                //quickFixes: {},
+            }
+            v.issues.push(issueO);
+
+            computeQuickFixesForIssue(issue, issue.entity, blueprintService, v.quickFixes)
+        });
+    });
```

Comment:
Duplicate code, can be moved into a function like:
```
export function computeQuickFixes(blueprintService, allIssues) {
    if (!allIssues) allIssues = blueprintService.getAllIssues();
    if (!allIssues) allIssues = {};
    if (!allIssues.errors) allIssues.errors = {};

    computeQuickFixesForIssues(allIssues.errors, blueprintService);
    computeQuickFixesForIssues(allIssues.warnings, blueprintService);
    
    return allIssues;
}

function computeQuickFixesForIssues(issuesCollection, blueprintService) {
    issuesCollection.byMessage = {};
    Object.values(issuesCollection.byEntity).forEach(list => {

    // etc.
```

Reply:
@algairim Yeah about that. This was the quickest and dirtiest way to make this work (the previous version supported only warning) 

However, the way the quick fixes are implemented doesn't really follow the AngularJS best practices or "way". I plan du refactor that which would make this bit of code go away. That's why I didn't go for a DRY implementation just now.

Reply:
Ok, a little effort to move this code into function though :)